### ISSUE TITLE
Dynamic Resource Decorator in Nest Js application 

### ISSUE THREAD

**Issue by hamzasaiff195** [ferrerojosh/nest-keycloak-connect#131]
We are using keycloak for user authentication and authorization in Nest Js application.
We want to implement resource-based-authorization and user-managed-access and for this purpose we need to customize or dynamic @Resource() Decorator.
For example if a person bob has resource named 'Alpha' then we have to pass that resources name in Decorator ==> @Resource('Alpha') then it'll allow request to move forward if this condition satisfied.
We don't want to hard code any value in @Resource Decorator we want it to be dynamic.

Any suggestions or solution will be appreciated.

**Comment by bonnzer** [comment]
Agree. Resource should accept dynamic data to allow pass resource names (or resource URIs #123, but names are preferred)
> Any suggestions or solution will be appreciated

There's no solution yet I guess

**Comment by ferrerojosh** [comment]
There is no solution for now, this needs to be a feature along with #123
